src/dagmc/DagMC.cpp

```diff
@@ -391,33 +390,51 @@ ErrorCode DagMC::create_graveyard(bool overwrite) {
   */
   ErrorCode rval;
 
-  // remove existing graveyard
+  // remove existing graveyard if overwrite is true
   if (overwrite) {
     remove_graveyard();
   }
 
-  // if a graveyard already exists at this point,
+  // if a graveyard already exists and we aren't overwriting it,
```

Comment:
If we are overwriting, then there shouldn't be a graveyard anymore.  Isn't this redundant?

Reply:
Also, doesn't the previous form of this catch a failure to remove the graveyard?

Reply:
Mmm yeah you're right. Thanks!